### ISSUE TITLE
Running on Matterport3D

### ISSUE THREAD

**Issue by Minzhao1995** [devendrachaplot/Object-Goal-Navigation#1]
Hi,

Thanks for your amazing works. I am trying to get it run on Matterport3D, but an info file (train_info.pbz2) is required. Would you be able to provide the codes for generating that file?

Thanks!

**Comment by PeihaoChen** [comment]
I am looking for the info file for MP3D, too.

Thanks!

**Comment by Minzhao1995** [comment]
I tried to generate the 'sem_map' using the object aabb, but it didn't perfectly match the map in the provided info file.
![Screenshot from 2021-04-16 12-05-59](https://user-images.githubusercontent.com/13643627/114970032-2d822700-9eac-11eb-9afe-722173c97dff.png)

**Comment by ggeorgak11** [comment]
In my understanding that file has information useful to estimate the object-nav metrics. If you can use directly the simulator instead, then you can ignore the train_info.pbz2

**Comment by Minzhao1995** [comment]
> In my understanding that file has information useful to estimate the object-nav metrics. If you can use directly the simulator instead, then you can ignore the train_info.pbz2

Yes, I directly used the simulator. But the problem here is how to determine the floor height, as one floor can span various heights in the MP3D dataset, and the floor height given by the level AABB annotation is unreliable.

**Comment by zenght** [comment]
> > In my understanding that file has information useful to estimate the object-nav metrics. If you can use directly the simulator instead, then you can ignore the train_info.pbz2
> 
> Yes, I directly used the simulator. But the problem here is how to determine the floor height, as one floor can span various heights in the MP3D dataset, and the floor height given by the level AABB annotation is unreliable.
I am also facing the same problem now. Is there an effective solution? Can we generate this info by ourselves?

**Comment by zenght** [comment]
> > In my understanding that file has information useful to estimate the object-nav metrics. If you can use directly the simulator instead, then you can ignore the train_info.pbz2
> 
> Yes, I directly used the simulator. But the problem here is how to determine the floor height, as one floor can span various heights in the MP3D dataset, and the floor height given by the level AABB annotation is unreliable.

I am also facing the same problem now. Is there an effective solution? Can we generate this info by ourselves

**Comment by ggeorgak11** [comment]
I don't think you need to generate this info if you are running experiments on MP3D. In get_metrics() at envs/habitat/objectgoal_env.py you can use directly the simulator to get the agent's position and distance from the goal positions. The goal positions can be found from the v1 dataset provided in the habitat-lab repository. 

You can verify this by tracing where the train_info.pbz2 info are being used in objectgoal_env.py. You can remove any use of that in the code. Also you don't need to pre-determine the floor-heights. If your input episodes contain start_position, start_rotation, object_category, and starting_distance (geodesic) then you should be fine.

**Comment by Minzhao1995** [comment]
> I don't think you need to generate this info if you are running experiments on MP3D. In get_metrics() at envs/habitat/objectgoal_env.py you can use directly the simulator to get the agent's position and distance from the goal positions. The goal positions can be found from the v1 dataset provided in the habitat-lab repository.
> 
> You can verify this by tracing where the train_info.pbz2 info are being used in objectgoal_env.py. You can remove any use of that in the code. Also you don't need to pre-determine the floor-heights. If your input episodes contain start_position, start_rotation, object_category, and starting_distance (geodesic) then you should be fine.

I see. I can use the v1 dataset directly. And this is a convenient way.  Thank you!

**Comment by zenght** [comment]
> I don't think you need to generate this info if you are running experiments on MP3D. In get_metrics() at envs/habitat/objectgoal_env.py you can use directly the simulator to get the agent's position and distance from the goal positions. The goal positions can be found from the v1 dataset provided in the habitat-lab repository.
> 
> You can verify this by tracing where the train_info.pbz2 info are being used in objectgoal_env.py. You can remove any use of that in the code. Also you don't need to pre-determine the floor-heights. If your input episodes contain start_position, start_rotation, object_category, and starting_distance (geodesic) then you should be fine.

Thanks, pal, but  I am still confused about how to get the goal position from the MP3Dv1 dataset provided in the habitat lab.

**Comment by zhangfuyang** [comment]
Hi. I am also facing the similar problem. I wonder how to reset the episode without train_info.pbz2 when using mp3d dataset? For gibson, we use ground truth `sem_map` from train_info.pbz2 to pick the new object in the new episode (function generate_new_episode), but how could this be done in mp3d?

Any suggestion would help! Thank you :)

**Comment by vincent341** [comment]
> train_info
Hi Min Zhao, would you mind giving more details about your solution?  How did you make it work on Mp3D? It would be very helpful to me. Thanks so much!

**Comment by jzhzhang** [comment]
> I don't think you need to generate this info if you are running experiments on MP3D. In get_metrics() at envs/habitat/objectgoal_env.py you can use directly the simulator to get the agent's position and distance from the goal positions. The goal positions can be found from the v1 dataset provided in the habitat-lab repository.
> 
> You can verify this by tracing where the train_info.pbz2 info are being used in objectgoal_env.py. You can remove any use of that in the code. Also you don't need to pre-determine the floor-heights. If your input episodes contain start_position, start_rotation, object_category, and starting_distance (geodesic) then you should be fine.

@ggeorgak11    Your advice is mostly correct and one can train the model with matterport. But there may have some mistakes. The metric, like ```distance_to_goal```, is not correct. Specifically, it seems to point to one target goal. You may have to build the dataset with the object_nav_dataset dataset. I'm still investigating this modification, and glad to hear more about your advice.

**Comment by BigTree765** [comment]
Could you give more details about your solution? How did you make it work on Mp3D? It would be very helpful to me. Thanks so much!
